.github/workflows/nightly-build.yaml

```diff
@@ -0,0 +1,24 @@
+name: Nightly Build
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
```

Comment:
Can we add a check to run the nightly build only if there have been changes in the last 24 hours? You may want to refer to [this](https://stackoverflow.com/a/67527144).